### PR TITLE
Add permissions for CRD

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

Follow up #243 

## How Has This Been Tested?

Deploying the actual code will fail, since the serviceaccount cannot list/update/get the new cards created. To test this fix.
1. Deploy latest image: podman pull quay.io/opendatahub/odh-dashboard:nightly
2. Access to the dashboard.
3. There won't be any application

Now build an image with this codebase and update the deployment:
1. Deploy it
2. Access to the dashboard.
3. You should see all the applications.

## Merge criteria:

- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: manifests/base/role.yaml [hunk 1]
```diff
@@ -101,3 +101,39 @@ rules:
...
+    resources:
+      - odhdocuments
+  - verbs:
+      - get
+      - list
+      - watch
+      - create
+      - update
+      - patch
+      - delete
```
Comment (andrewballantyne) [inline manifests/base/role.yaml:135]: Are we sure we want to grant full access and not just read access?

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: manifests/base/role.yaml (hunk 1 above)
Comment (dlabaj) [reply manifests/base/role.yaml:135]: I think read will be fine.

**Review by dlabaj — APPROVED** [review]
/lgtm

File: manifests/base/role.yaml (hunk 1 above)
Comment (andrewballantyne) [reply manifests/base/role.yaml]: FWIW, `watch` could have been in here too... but since we don't use watches today, it probably isn't a concern.

**Pull request merged**